diff-pdf.cpp

```diff
@@ -386,6 +386,12 @@ bool doc_compare(PopplerDocument *doc1, PopplerDocument *doc2,
                        );
         }
 
+        poppler_page_get_size(poppler_document_get_page(doc1, page), &w, &h);
```

Comment:
This should be inside that `if` statement underneath as well — no point in doing some work that you then won't use.

Reply:
I will change it, but I have to add:
If in the future someone would use/need w and h for something else, the value would only be updated if pdf_output is true.

Implying: the variables w and h are declared in the wrong scope.
We can not assume the width and height of a pdf do be the same for all pages.

Either we update it for every page - or - to avoid needlessly requesting it, reduce the variable scopes. As you imply, you prefer not to needlessly create work.

Commit will follow - I think that will make my thinking clearer.

Reply:
> Implying: the variables w and h are declared in the wrong scope.

Yep. I didn't want to burden you larger changes, so I didn't say anything, but I agree 100% with you and your subsequent change is best — thanks a lot! I prefer it this way instead of always updating "global" `w` and `h` variables because it is easy to miss when they are used. This way, with them locally scoped, if some future change needs them for something, one will be forced to think about the implications.

Reply:
Thank you for your feedback.

I have no issues of going the extra mile, when it means that we get a more stable software.
I want to migrate from my custom GIMP script to this software, as your solution/software is in every aspect faster then my current solution. But being fast only helps when the software is reliable (in the context of my use case).

Feel free to squash the changes, so that you have a clean git history.